bigframes/core/compile/compiled.py

```diff
@@ -477,29 +487,148 @@ def project_ternary_op(
         col_id_3: str,
         op: ops.TernaryOp,
         output_column_id: str,
-    ) -> CompiledArrayValue:
-        """Creates a new expression based on this expression with ternary operation applied to three columns."""
+    ) -> UnorderedIR:
         value = op(
```

Comment:
Looks like we can share some of these implementations across unordered / ordered with a `_maybe_hide_columns` or something "template pattern".

Reply:
Templated out to an abstract `_set_or_replace_by_id` method that assigns ibis values to column ids.